packages/vm/src/cache.rs

```diff
@@ -116,16 +116,16 @@ where
         backend: Backend<A, S, Q>,
         options: InstanceOptions,
     ) -> VmResult<Instance<A, S, Q>> {
-        let store = make_runtime_store(options.memory_limit);
         // Get module from memory cache
-        if let Some(module) = self.memory_cache.load(checksum, &store)? {
+        if let Some(module) = self.memory_cache.load(checksum)? {
```

Comment:
In this case, `options.memory_limit` is not used. Instead any unknown memory limit from the time of storing to cache is used. For this reason, the `memory_limit` should not be a parameter of `::get_instance` anymore. Instead it can me a member of `Cache`

Reply:
A `store` is still needed below, to deserialize from disk.

But I agree, removing the memory limit paremeter from `get_instance` is a good idea. We can create an internal store instance per `Cache` directly, and use it when needed.

Reply:
> A `store` is still needed below, to deserialize from disk.

Right, this makes the API even more problematic. Sometimes the limit is applied and sometimes it is not.

> We can create an internal store instance per Cache directly, and use it when needed.

This is a good idea in general. However, only one `Metering` instance can be used per store (Due to some internal reasons. It will panic if you use une store for multiple `Metering` instances). I.e. every `Module` needs its own `Store` in our use case.